packages/extension/src/view/popup/stateProviders/cookie/cookieProvider.tsx

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies.
+ */
+import React, {
+  type PropsWithChildren,
+  useEffect,
+  useState,
+  useCallback,
+  useRef,
+} from 'react';
+import { useDebouncedCallback } from 'use-debounce';
+import { prepareCookiesCount } from '@ps-analysis-tool/design-system';
+import { type CookieData } from '@ps-analysis-tool/common';
+
+/**
+ * Internal dependencies.
+ */
+import { getCurrentTab } from '../../../../utils/getCurrentTabId';
+import {
+  ALLOWED_NUMBER_OF_TABS,
+  INITIAL_SYNC,
+  NEW_COOKIE_DATA,
+  POPUP_CLOSE,
+  POPUP_OPEN,
+  SERVICE_WORKER_RELOAD_MESSAGE,
+  SET_TAB_TO_READ,
+} from '../../../../constants';
+import { useSettings } from '../settings';
+import Context, { type CookieStoreContext } from './context';
+
+const Provider = ({ children }: PropsWithChildren) => {
+  const [tabId, setTabId] = useState<number | null>(null);
+
+  const [tabToRead, setTabToRead] = useState<string | null>(null);
+  const [tabCookieStats, setTabCookieStats] =
+    useState<CookieStoreContext['state']['tabCookieStats']>(null);
+
+  const [returningToSingleTab, setReturningToSingleTab] =
+    useState<CookieStoreContext['state']['returningToSingleTab']>(false);
+
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const [isCurrentTabBeingListenedTo, setIsCurrentTabBeingListenedTo] =
+    useState<boolean>(false);
+
+  const [onChromeUrl, setOnChromeUrl] = useState<boolean>(false);
+
+  const loadingTimeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const { setSettingsChanged } = useSettings(({ actions }) => ({
+    setSettingsChanged: actions.setSettingsChanged,
+  }));
+
+  useEffect(() => {
+    loadingTimeout.current = setTimeout(() => {
+      setLoading(false);
+    }, 6500);
+
+    return () => {
+      if (loadingTimeout.current) {
+        clearTimeout(loadingTimeout.current);
+      }
+    };
+  }, []);
+
+  const setDebouncedStats = useDebouncedCallback((value) => {
```

Comment:
I guess it's not used now, can be removed if true.

Reply:
We still use it in the storage listener.